supervision/metrics/detection.py

```diff
@@ -0,0 +1,420 @@
+            ):
+                raise ValueError(
+                    f"Predictions and targets must be lists of numpy arrays. Got {type(predictions[0])} and {type(targets[0])} instead."
+                )
+            if predictions[0].shape[1] != 6:
+                raise ValueError(
+                    f"Predictions must have shape (N, 6). Got {predictions[0].shape} instead."
+                )
+            if targets[0].shape[1] != 5:
+                raise ValueError(
+                    f"Targets must have shape (N, 5). Got {targets[0].shape} instead."
+                )
+
+    @staticmethod
+    def _evaluate_detection_batch(
+        true_detections: np.ndarray,
+        pred_detections: np.ndarray,
+        num_classes: int,
+        conf_threshold: float,
+        iou_threshold: float,
+    ) -> np.ndarray:
+        """
+        Calculate confusion matrix for a batch of detections for a single image.
+
+        Args:
+            See ConfusionMatrix.from_detections
+
+        Returns:
+            confusion matrix based on a single image.
+        """
+        result_matrix = np.zeros((num_classes + 1, num_classes + 1))
+
+        conf_idx = 5
+        confidence = pred_detections[:, conf_idx]
+        detection_batch_filtered = pred_detections[confidence > conf_threshold]
+
+        class_id_idx = 4
+        true_classes = np.array(true_detections[:, class_id_idx], dtype=np.int16)
+        detection_classes = np.array(
+            detection_batch_filtered[:, class_id_idx], dtype=np.int16
+        )
+        true_boxes = true_detections[:, :class_id_idx]
+        detection_boxes = detection_batch_filtered[:, :class_id_idx]
+
+        iou_batch = box_iou_batch(
+            boxes_true=true_boxes, boxes_detection=detection_boxes
+        )
+        matched_idx = np.asarray(iou_batch > iou_threshold).nonzero()
+
+        if matched_idx[0].shape[0]:
+            matches = np.stack(
+                (matched_idx[0], matched_idx[1], iou_batch[matched_idx]), axis=1
+            )
+            matches = ConfusionMatrix._drop_extra_matches(matches=matches)
+        else:
+            matches = np.zeros((0, 3))
+
+        matched_true_idx, matched_detection_idx, _ = matches.transpose().astype(
+            np.int16
+        )
+
+        for i, true_class_value in enumerate(true_classes):
+            j = matched_true_idx == i
+            if matches.shape[0] > 0 and sum(j) == 1:
+                result_matrix[
+                    true_class_value, detection_classes[matched_detection_idx[j]]
+                ] += 1  # TP
+            else:
+                result_matrix[true_class_value, num_classes] += 1  # FN
+
+        for i, detection_class_value in enumerate(detection_classes):
+            if not any(matched_detection_idx == i):
+                result_matrix[num_classes, detection_class_value] += 1  # FP
+
+        return result_matrix
+
+    @staticmethod
+    def _drop_extra_matches(matches: np.ndarray) -> np.ndarray:
+        """
```

Comment:
Let's drop the docstring here. We have a script that processes our code and builds documentation. It is looking for docstings. Given the fact that this method is private, we wouldn't want it to be exposed.

Reply:
_evaluate_detection_batch and _drop_extra_matches have docstrings that shouldn't be exposed, yet the methods don't seem straightforward to parse quickly with a human eye. Is there a solution on the documentation builder's side to skip methods that start with an underscore?

Reply:
I agree with making _evaluate_detection_batch public, but the question from above still holds

Reply:
You can get around it with the regular Python comment `#` at the top of the method body.